oximeter/collector/src/agent.rs

```diff
@@ -70,22 +72,30 @@ pub struct OximeterAgent {
 
 impl OximeterAgent {
     /// Construct a new agent with the given ID and logger.
+    // TODO: Remove this linter exception once we only write to a
+    // single database
+    #[allow(clippy::too_many_arguments)]
     pub async fn with_id(
         id: Uuid,
         address: SocketAddrV6,
         refresh_interval: Duration,
         db_config: DbConfig,
         native_resolver: BoxedResolver,
+        // Temporary resolver to write to a replicated ClickHouse
+        // cluster as well as a single-node installation.
+        cluster_resolver: BoxedResolver,
         log: &Logger,
         replicated: bool,
     ) -> Result<Self, Error> {
-        let (result_sender, result_receiver) = mpsc::channel(8);
+        let collection_task_wrapper = CollectionTaskWrapper::new();
+
         let log = log.new(o!(
             "component" => "oximeter-agent",
             "collector_id" => id.to_string(),
             "collector_ip" => address.ip().to_string(),
         ));
         let insertion_log = log.new(o!("component" => "results-sink"));
+        let instertion_log_cluster = insertion_log.clone();
```

Comment:
Could we use a different component for this, rather than cloning the existing log? It might help diagnose different behavior between the single-node and cluster.

Reply:
Done in [95db1da](https://github.com/oxidecomputer/omicron/pull/7565/commits/95db1dac3725a65ee3c373ad40b40cf4e19e549f)